### PR TITLE
[Enhancement] Some code refactor for TDE

### PR DESCRIPTION
**Pull request by decster**
## Why I'm doing:

Some TDE related code needs refactor for better extension and testing 

## What I'm doing:

Some minor refactor of TDE code

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/fs/key_cache.cpp [hunk 1]
```diff
@@ -272,7 +270,9 @@ Status KeyCache::refresh_keys(const std::string& key_meta) {
     std::vector<const EncryptionKey*> keys(nkey);
     std::vector<std::unique_ptr<EncryptionKey>> owned_keys(nkey);
     RETURN_IF_ERROR(_resolve_encryption_meta(meta_pb, keys, owned_keys, nkey - 1));
-    LOG(INFO) << "refresh keys, num keys before: " << size_before << " after:" << size();
+    if (size_before != size()) {
+        LOG(INFO) << "refresh keys, num keys before: " << size_before << " after:" << size();
```
Comment (wyb) [inline be/src/fs/key_cache.cpp:274]: WARNING?

**Review by wyb — DISMISSED** [review]

File: be/src/fs/key_cache.cpp (hunk 1 above)
Comment (decster) [reply be/src/fs/key_cache.cpp:274]: no, it's normal

File: be/src/util/c_string.h [hunk 2]
```diff
@@ -14,6 +14,8 @@
 
 #pragma once
 
+#include <stdint.h>
```
Comment (kevincai) [inline be/src/util/c_string.h:17]: already included in https://github.com/StarRocks/starrocks/pull/49351/files#diff-8263193bc080d7ba8dc61bf8414b552be86a6cb0c5ba8fc88c8dce147a343add

this line can be removed now.

Comment (decster) [reply be/src/util/c_string.h]: done

**Review by kevincai — APPROVED** [review]

**Review by alvin-phoenix-ai — APPROVED** [review]

**Pull request merged**